psi4/src/psi4/libfock/DFJLinK.cc

```diff
@@ -0,0 +1,932 @@
+    timer_on("DFJLinK: ERI Computers");
+    
+    auto zero = BasisSet::zero_ao_basis_set();
+    
+    // initialize 4-Center ERIs  
+    eri_computers_["4-Center"].emplace({}); 
+    eri_computers_["4-Center"].resize(nthreads_);
+    
+    IntegralFactory factory(primary_, primary_, primary_, primary_);
+    eri_computers_["4-Center"][0] = std::shared_ptr<TwoBodyAOInt>(factory.eri());
+    
+    // initialize 3-Center ERIs
+    eri_computers_["3-Center"].emplace({});
+    eri_computers_["3-Center"].resize(nthreads_);
+    
+    IntegralFactory rifactory(auxiliary_, zero, primary_, primary_);
+    eri_computers_["3-Center"][0] = std::shared_ptr<TwoBodyAOInt>(rifactory.eri());
+    
+    // create each threads' ERI computers 
+    for(int rank = 1; rank < nthreads_; rank++) {
+        eri_computers_["4-Center"][rank] = std::shared_ptr<TwoBodyAOInt>(eri_computers_["4-Center"].front()->clone());
+        eri_computers_["3-Center"][rank] = std::shared_ptr<TwoBodyAOInt>(eri_computers_["3-Center"].front()->clone());
+    }
+
+    timer_off("DFJLinK: ERI Computers");
+
+    // => Direct Density-Fitted Coulomb Setup <= //
+
+    // pre-compute coulomb fitting metric
+    timer_on("DFJLinK: Coulomb Metric");
+    
+    FittingMetric J_metric_obj(auxiliary_, true);
+    J_metric_obj.form_fitting_metric();
+    J_metric_ = J_metric_obj.get_metric();
+    
+    timer_off("DFJLinK: Coulomb Metric");
+
+    // => Linear Exchange Setup <= //
+    
+    // set up LinK integral tolerance
+    if (options_["LINK_INTS_TOLERANCE"].has_changed()) {
+        linK_ints_cutoff_ = options_.get_double("LINK_INTS_TOLERANCE");
+    } else {
+        linK_ints_cutoff_ = cutoff_; 
+    }
+}
+size_t DFJLinK::num_computed_shells() { 
+    //no bench data returned - to come in a future update
+    return JK::num_computed_shells(); 
+}
+
+size_t DFJLinK::memory_estimate() {
+    return 0;  // Memory is O(N^2), which psi4 counts as effectively 0
+}
+
+void DFJLinK::print_header() const {
+    std::string screen_type = options_.get_str("SCREENING");
+    if (print_) {
+        outfile->Printf("  ==> DFJLinK: Density-Fitted J and Linear Exchange K <==\n\n");
+
+        outfile->Printf("    J tasked:           %11s\n", (do_J_ ? "Yes" : "No"));
+        outfile->Printf("    K tasked:           %11s\n", (do_K_ ? "Yes" : "No"));
+        outfile->Printf("    wK tasked:          %11s\n", (do_wK_ ? "Yes" : "No"));
+        if (do_wK_) outfile->Printf("    Omega:              %11.3E\n", omega_);
+        outfile->Printf("    Integrals threads:  %11d\n", nthreads_);
+        outfile->Printf("    Memory [MiB]:       %11ld\n", (memory_ *8L) / (1024L * 1024L));
+        outfile->Printf("    Incremental Fock :  %11s\n", (incfock_ ? "Yes" : "No"));
+        outfile->Printf("    Screening Type:   %11s\n", screen_type.c_str());
+        outfile->Printf("    J Screening Cutoff: %11.0E\n", cutoff_);
+        outfile->Printf("    K Screening Cutoff: %11.0E\n", linK_ints_cutoff_);
```

Comment:
I put some comments in one of the output.ref's. They belong here. :-)

Reply:
So I think I may have missed the commit where the comments in the output.ref were added.

Reply:
The output.refs look good to me now, so whatever happened, I think it's resolved. Thanks for the fix.

Reply:
All right, excellent; thanks!